### ISSUE TITLE
Inventory Overhaul

### ISSUE THREAD

**Issue by Jacki3** [LoGaCulture/LUTE#82]
**Is your feature request related to a problem? Please describe.**
Current inventory system relies on some other packages that are either reliant on some other assembly def or are tied too closely to LUTE and therefore cannot be applied easily to other scenarios (they are not fully standalone). We require the inventory system to be somewhat overhauled just as the Achievement system was.

**Additional context**
A simple, grid-based inventory system that also improves:

1. The scrolling nature (optional add-on) for a larger inventory system with lots and lots of slots (either vertical or horizontal).
2. Ensuring that items and inventory status are accessible and serialisable with ease (a new saving option to be added as we did for the Achievement system).
2. Ensuring that inventory items only need to be placed in 'Resources' folders rather than sub-folders inside this one.
3. A simplified approach that streamlines the UX and UI making implementation of the system into other systems a lot easier and not reliant on LUTE (using specific listeners to be a bit more generic).

Please refer to the current inventory system for a base and then updated related Orders to match this new system. Upon successful rewrite, all prior references to the old inventory system should be removed (including the source files).

**Comment by Jacki3** [comment]
Marking this issue as closed as the Inventory has been complete. You can find more information here:

https://github.com/LoGaCulture/LUTE/wiki/Inventory

In addition please see the changelog and commits speicfically related to this issue/feature request.
